syntax = "proto3";

package ac;

message Point {
  int32 x = 1;
  int32 y = 2;
}

message RelativePoint {
  int32 x = 1;
  int32 y = 2;
  int32 width = 3;
  int32 height = 4;
}

/**
 * Represents a rectangular region.
 * Meant to be used as the region [left,right] x [top,bottom] (inclusive).
 *
 * When used for clickable regions (aka repeated Rect):
 * - This can represent a point (when left/top coincide with right/bottom)
 * - This can represent a region (single Rect)
 * - This can represent composite region (repeated Rect)
 */
message Rect {
  int32 left = 1;
  int32 right = 2;
  int32 top = 3;
  int32 bottom = 4;
}

message PointSequence {
  repeated Point points = 1;
}
